### PR TITLE
Tolerate structural whitespace in EEBUS JSON

### LINKED ISSUE #97: Bug: TC_SHIP_MSG_003 — `JsonFromEEBUSJson` breaks on whitespace-formatted EEBUS JSON
**Test case:** TC_SHIP_MSG_003 (spec §4.5.3)
**Requirement:** SHIP-TS-MSG-03 → JSON specification (RFC 8259 §2)
**Severity:** FAIL (mandatory)
**DUT role:** Server & client

---

## Spec requirement

SHIP test spec §4.5.3:

> The DUT MUST correctly parse SHIP messages containing structural whitespace (Space 0x20,
> Tab 0x09, LF 0x0A, CR 0x0D) before or after structural characters `{`, `}`, `[`, `]`,
> `:`, `,` as permitted by the JSON specification (RFC 8259 §2).

TC_SHIP_MSG_003 test scenario:

1. Test tool sends a valid SHIP message whose EEBUS JSON payload contains whitespace around
   structural tokens (e.g. `[ {"foo":"bar"} ]` or `[{"a":1}, {"b":2}]`)
2. Expected: DUT parses the message without error and processes it identically to the
   compact equivalent
3. Expected outcome within 30 s: DUT responds correctly, no connection tear-down

---

## Current implementation

`ship/helper.go:12–20` — `JsonFromEEBUSJson()`:

```go
func JsonFromEEBUSJson(json []byte) []byte {
    var result = bytes.ReplaceAll(json, []byte("[{"), []byte("{"))
    result = bytes.ReplaceAll(result, []byte("},{"), []byte(","))
    result = bytes.ReplaceAll(result, []byte("}]"), []byte("}"))
    result = bytes.ReplaceAll(result, []byte("[]"), []byte("{}"))
    result = bytes.Trim(result, "\x00")
    return result
}
```

`ship/connection_serialization.go:144` — `parseMessage()` passes raw bytes directly to
`JsonFromEEBUSJson` with no whitespace normalisation beforehand:

```go
if jsonFormat {
    return shipHeaderByte, JsonFromEEBUSJson(msg)
}
```

---

## Why TC_SHIP_MSG_003 fails

`JsonFromEEBUSJson` assumes no whitespace exists outside of string values. The replacer
uses exact byte-sequence matching, so any structural whitespace between tokens breaks the
match, leaving the outer array intact. The unconverted byte slice is then
passed to `json.Unmarshal` as a SHIP message, which fails to unmarshal into the expected Go
struct (outer array instead of object).

| Input | Expected output | Actual output |
|---|---|---|
| `[{"foo":"bar"}]` | `{"foo":"bar"}` | `{"foo":"bar"}` ✓ |
| `[ {"foo":"bar"} ]` | `{"foo":"bar"}` | `[ {"foo":"bar"} ]` ✗ |
| `[{"a":1},{"b":2}]` | `{"a":1,"b":2}` | `{"a":1,"b":2}` ✓ |
| `[{"a":1}, {"b":2}]` | `{"a":1,"b":2}` | `[{"a":1}, {"b":2}]` ✗ |
| `[{"a":1} ,{"b":2}]` | `{"a":1,"b":2}` | `[{"a":1} ,{"b":2}]` ✗ |

---

## Affected code locations

| File | Location | Description |
|---|---|---|
| `ship/helper.go` | `JsonFromEEBUSJson()` L12–20 | Byte-exact replacement breaks on whitespace-formatted EEBUS JSON |
| `ship/connection_serialization.go` | `parseMessage()` L144 | Caller — passes raw bytes directly to `JsonFromEEBUSJson` without normalisation |
| `ship/helper_test.go` | `TestJsonFromEEBUSJson` L11 | Tests only cover compact (non-whitespace) inputs |

---

## Required fix

Replace byte-exact matching with whitespace-tolerant parsing. Options:

1. **Regex-based** — replace `\[\s*\{` → `{`, `\}\s*,\s*\{` → `,`, `\}\s*\]` → `}`,
   `\[\s*\]` → `{}`
2. **JSON unmarshal/remarshal** — unmarshal into `[]json.RawMessage`, then remarshal as
   object — fully spec-compliant but heavier

Also add test cases with whitespace-formatted inputs to `ship/helper_test.go`.

TestSpec Analysis done by me, ticket description done by Claude. Detailed reasoning and required fix may require further manual analysis.

### PR DESCRIPTION
**Pull request by kirollosnct**
JsonFromEEBUSJson unwraps the EEBUS array-of-single-key-objects encoding via exact byte-sequence replacements. Any RFC 8259 structural whitespace around JSON tokens defeated those matches, leaving the outer array intact so the downstream json.Unmarshal failed.

Normalise the input with json.Compact before the replacements. Compact is string-aware, so whitespace inside string values is preserved; invalid JSON falls back to the raw bytes so behaviour is no worse than before. The NUL trim moves ahead of Compact since a trailing 0x00 is invalid JSON.

Add whitespace-tolerance tests covering all four whitespace characters around structural tokens, whitespace inside string values, and a pretty-printed real datagram.

Fixes #97.

### CODE REVIEW

File: ship/helper.go [hunk 1]
```diff
@@ -9,13 +9,21 @@ import (
...
+func JsonFromEEBUSJson(data []byte) []byte {
+	// The PMCP device mistakenly adds an `0x00` byte at the end of many
+	// messages. Trim it first so the remaining bytes are valid JSON.
+	data = bytes.Trim(data, "\x00")
+
+	// Compact JSON to remove extra whitespaces
+	var compacted bytes.Buffer
+	if err := json.Compact(&compacted, data); err == nil {
+		data = compacted.Bytes()
+	}
```
Comment (sthelen-enqs) [inline ship/helper.go:21]: if err != nil (i.e. if the input is not valid JSON), should we just return data here without running the replacements first?
I guess it doesn't really matter since we'll throw an error when unmarshalling anyway but it might be nice to have a comment here describing what happens if/when Compact fails.
